### PR TITLE
Open up SonatypeHost

### LINKED ISSUE #503: How to set a custom host in publishToMavenCentral?
```
fun publishToMavenCentral(host: SonatypeHost = SonatypeHost.DEFAULT, automaticRelease: Boolean = false) {
```

SonatypeHost is an enum, so how to set a custom host in the form of xxx.oss.sonatype.org for publishing to Sonetype?

DEFAULT and use SONATYPE_HOST env? Can the enum be opened up to a value type?

### PR DESCRIPTION
**Pull request by yschimke**
Relates to https://issues.sonatype.org/browse/OSSRH-88192

Make SonatypeHost a value class instead of an enum.

fixes #503

### CODE REVIEW

**Comment by yschimke** [comment]
Testing failed

`org.gradle.internal.snapshot.ValueSnapshottingException: Could not serialize value of type SonatypeHost`

**Comment by yschimke** [comment]
Now working at least to the point of a 403.

```
Caused by: java.io.IOException: Cannot get stagingProfiles for account yuri@schimke.ee: <html>
  <head>
    <title>403 - Forbidden</title>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

    <link rel="icon" type="image/png" href="https://google.oss.sonatype.org/favicon.png">
    <!--[if IE]>
    <link rel="SHORTCUT ICON" href="https://google.oss.sonatype.org/favicon.ico"/>
    <![endif]-->
```

File: plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt
Comment (gabrielittner) [inline plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt:38]: Could you add `@JvmStatic` and `@JvmField` to the function and properties respectively? That should keep compatibility for Groovy users.

**Review by gabrielittner — COMMENTED** [review]
I think we also need an overload like this in `MavenPublishBaseExtension`: 
```
@JvmOverloads
fun publishToMavenCentral(host: String = "DEFAULT" automaticRelease: Boolean = false) 
```
Groovy (or maybe just Gradle) has something that allows passing an enum value as string to a function.

**Comment by yschimke** [comment]
> I think we also need an overload like this in `MavenPublishBaseExtension`: 
> ```
> @JvmOverloads
> fun publishToMavenCentral(host: String = "DEFAULT" automaticRelease: Boolean = false) 
> ```
> Groovy (or maybe just Gradle) has something that allows passing an enum value as string to a function.

I'm not confident about this part of the change. Won't the overload conflict when no params are provided? Like `publishToMavenCentral()`

**Comment by yschimke** [comment]
Feel free to takeover this PR, or suggest further changes.

File: plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt
Comment (yschimke) [inline plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt:84]: Should I use @ JvmSynthetic to hide from Kotlin?

Comment (gabrielittner) [reply plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt:84]: That's a good idea.

We can also leave out the default value for `host` to avoid your concern about conflicting overloads

**Comment by gabrielittner** [comment]
Looks like there is a ktlint issue, can you run `ktlintFormat` once? Otherwise I can also do that before merging
```
> Task :plugin:ktlintMainSourceSetCheck FAILED
/home/runner/work/gradle-maven-publish-plugin/gradle-maven-publish-plugin/plugin/src/main/kotlin/com/vanniktech/maven/publish/SonatypeHost.kt:14:2 Missing spacing before ":"
```

**Review by gabrielittner — APPROVED** [review]
Thanks for the contribution. I can create a new release with this soon, but we're also blocked by the Sonatype issues 😬

**Pull request merged**
